templates/cronjob.yaml

```diff
@@ -0,0 +1,222 @@
+                    key: s3SecretKey
+              {{- end }}
+              - name: REGISTRY_STORAGE_S3_REGION
+                value: {{ required ".Values.s3.region is required" .Values.s3.region }}
+            {{- if .Values.s3.regionEndpoint }}
+              - name: REGISTRY_STORAGE_S3_REGIONENDPOINT
+                value: {{ .Values.s3.regionEndpoint }}
+            {{- end }}
+              - name: REGISTRY_STORAGE_S3_BUCKET
+                value: {{ required ".Values.s3.bucket is required" .Values.s3.bucket }}
+            {{- if .Values.s3.rootdirectory }}
+              - name: REGISTRY_STORAGE_S3_ROOTDIRECTORY
+                value: {{ .Values.s3.rootdirectory | quote }}
+            {{- end }}
+            {{- if .Values.s3.encrypt }}
+              - name: REGISTRY_STORAGE_S3_ENCRYPT
+                value: {{ .Values.s3.encrypt | quote }}
+            {{- end }}
+            {{- if .Values.s3.secure }}
+              - name: REGISTRY_STORAGE_S3_SECURE
+                value: {{ .Values.s3.secure | quote }}
+            {{- end }}
+{{- else if eq .Values.storage "swift" }}
+              - name: REGISTRY_STORAGE_SWIFT_AUTHURL
+                value: {{ required ".Values.swift.authurl is required" .Values.swift.authurl }}
+              - name: REGISTRY_STORAGE_SWIFT_USERNAME
+                valueFrom:
+                  secretKeyRef:
+                    name: {{ template "docker-registry.fullname" . }}-secret
+                    key: swiftUsername
+              - name: REGISTRY_STORAGE_SWIFT_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: {{ template "docker-registry.fullname" . }}-secret
+                    key: swiftPassword
+              - name: REGISTRY_STORAGE_SWIFT_CONTAINER
+                value: {{ required ".Values.swift.container is required" .Values.swift.container }}
+{{- end }}
+{{- if .Values.proxy.enabled }}
+              - name: REGISTRY_PROXY_REMOTEURL
+                value: {{ required ".Values.proxy.remoteurl is required" .Values.proxy.remoteurl }}
+              - name: REGISTRY_PROXY_USERNAME
+                valueFrom:
+                  secretKeyRef:
+                    name: {{ if .Values.proxy.secretRef }}{{ .Values.proxy.secretRef }}{{ else }}{{ template "docker-registry.fullname" . }}-secret{{ end }}
+                    key: proxyUsername
+              - name: REGISTRY_PROXY_PASSWORD
+                valueFrom:
+                  secretKeyRef:
+                    name: {{ if .Values.proxy.secretRef }}{{ .Values.proxy.secretRef }}{{ else }}{{ template "docker-registry.fullname" . }}-secret{{ end }}
+                    key: proxyPassword
+{{- end }}
+{{- if .Values.persistence.deleteEnabled }}
+              - name: REGISTRY_STORAGE_DELETE_ENABLED
+                value: "true"
+{{- end }}
```

Comment:
💡 **Note:** This environment variable (and corresponding value in values.yaml) is introduced in this PR, so in subsequent refactoring, just need to be careful of regression.